dtfs-central-api/src/utils/report-period.ts

```diff
@@ -27,3 +29,45 @@ export const getPreviousReportPeriodStart = ({ month, year }: MonthAndYear): Mon
 
 export const isEqualReportPeriodStart = (reportPeriodStart1: MonthAndYear, reportPeriodStart2: MonthAndYear): boolean =>
   reportPeriodStart1.year === reportPeriodStart2.year && reportPeriodStart1.month === reportPeriodStart2.month;
+
+export const getCurrentReportPeriodForBankSchedule = (bankReportPeriodSchedule: ReportPeriodSchedule[]): ReportPeriod => {
+  const currentReportPeriodDate = subMonths(new Date(), 1);
+  const oneIndexedPreviousMonth = getOneIndexedMonth(currentReportPeriodDate);
+  const currentYear = currentReportPeriodDate.getFullYear();
+
+  const reportPeriodInPreviousYear = bankReportPeriodSchedule.find(
+    (schedule) => schedule.startMonth > schedule.endMonth && (oneIndexedPreviousMonth >= schedule.startMonth || oneIndexedPreviousMonth <= schedule.endMonth),
+  );
+  if (reportPeriodInPreviousYear) {
+    return {
+      start: {
+        month: reportPeriodInPreviousYear.startMonth,
+        year: currentYear - 1,
+      },
+      end: {
+        month: reportPeriodInPreviousYear.endMonth,
+        year: currentYear,
+      },
+    };
```

Comment:
Following my comment about naming, is this logic correct?

Reply:
Not for that specific edge case, no. I will add a new test to cover it